src/jit/emitxarch.cpp

```diff
@@ -702,6 +715,19 @@ unsigned emitter::emitOutputRexOrVexPrefixIfNeeded(instruction ins, BYTE* dst, c
         //     VEX.0011RM22 got transformed as VEX.0000RM22
         //
         // Now output VEX prefix leaving the 4-byte opcode
+
+        if (Supports2ByteVexEncoding(ins) && ((vexPrefix & 0xFFFF7F80) == 0x00C40000))
```

Comment:
I had initially started doing this by updating `AddVexPrefix` to support both the 2 and 3-byte encodings. However, that was leading to a cascade of changes and required knowledge of things like: "is the instruction using any extended registers".

I ended up realizing that, at the point we actually emit the encoding, we have all the relevant information and can just make the decision there, so this ended up being significantly simpler to support than initially expected.